plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPlugin.kt

```diff
@@ -73,16 +84,57 @@ open class KotlinDslPlugin : Plugin<Project> {
     }
 
     private
-    fun Project.configureCompilerTasks() {
+    fun Project.enablePrecompiledScripts() {
         tasks {
-            "compileKotlin"(KotlinCompile::class) {
-                kotlinOptions {
-                    freeCompilerArgs += listOf("-script-templates", scriptTemplates)
+
+            val implicitImportsFile = kotlinDslBuildDir().map { it.file("implicit-imports") }
+            val generateImplicitImportsFile by creating {
```

Comment:
❓ this task will be executed for each source set that contains precompiled scripts. But the content of this file is bound to `(Gradle.version + KotlinDsl.version)`. Couldn't we generate this import file in some cache with a wider scope instead?